ParseStarterProject/IntroViewController.swift

```diff
@@ -0,0 +1,119 @@
+            // Set up the sign up view controller
+            let signUpLogoTitle = UILabel()
+            signUpLogoTitle.text = Constants.appName
+            self.signUpViewController.signUpView?.logo = signUpLogoTitle
+            self.signUpViewController.delegate = self
+            self.loginViewController.signUpController = self.signUpViewController
+            // We have no current user, so let's show them the login view
+            continueToLogin()
+        } else {
+            self.continueToMainApp()
+        }
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+    // ----- Mark: PFLogInViewControllerDelegate
+    
+    func logInViewController(logInController: PFLogInViewController, shouldBeginLogInWithUsername username: String, password: String) -> Bool {
+        if (username.isEmpty || password.isEmpty) {
+            return false
+        }
+        return true
+    }
+    
+    func logInViewController(logInController: PFLogInViewController, didLogInUser user: PFUser) {
+        continueToMainApp()
+    }
+    
+    func logInViewController(logInController: PFLogInViewController, didFailToLogInWithError error: NSError?) {
+        print("Failed to login")
+        // TODO: Error Handling
+    }
+    
+    func logInViewControllerDidCancelLogIn(logInController: PFLogInViewController) {
+        print("Dismissed login")
+    }
+    
+    // ----- Mark: PFSignUpViewControllerDelegate
+    
+    func signUpViewController(signUpController: PFSignUpViewController, shouldBeginSignUp info: [NSObject : AnyObject]) -> Bool {
+        let username = info["username"] as! String
+        let password = info["password"] as! String
+        let email = info["email"] as! String
+        if (username.isEmpty || password.isEmpty || email.isEmpty) {
+            return false
+        }
+        return true
+    }
+    
+    func signUpViewController(signUpController: PFSignUpViewController, didSignUpUser user: PFUser) {
+        continueToMainApp()
+    }
+    
+    func signUpViewController(signUpController: PFSignUpViewController, didFailToSignUpWithError error: NSError?) {
+        print("Failed to signup")
+        // TODO: Error Handling
+    }
+    
+    func signUpViewControllerDidCancelSignUp(signUpController: PFSignUpViewController) {
+        print("User dismissed sign up")
+    }
+    
+    // ----- Mark: View transitions
+    
+    func continueToLogin() {
+        self.presentViewController(loginViewController, animated: true, completion: nil)
+    }
+    
+    /*
+    * User is signed up or logged in. Lets take them to the main app
+    */
+    func continueToMainApp() {
+        //let storyboard = Constants.storyboard
+        //let mainAppVC = storyboard.instantiateViewControllerWithIdentifier("nameOfVCGoesHere")
+        //self.presentViewController(mainAppVC, animated: true, completion: nil)
+    }
```

Comment:
Not good practice to push code that is commented out, but I didn't have the next view controller set up yet and wanted to include this logic in this commit.